[Tests] RDS Global cluster fails to destroy when `force_destroy` is set

### Description

When trying to destroy a RDS global cluster it fails if `force_destroy` is set and there were no RDS clusters associated with it

```
=== RUN   TestAccRDSCluster_GlobalClusterIdentifierEngineModeGlobal_add
=== PAUSE TestAccRDSCluster_GlobalClusterIdentifierEngineModeGlobal_add
=== CONT  TestAccRDSCluster_GlobalClusterIdentifierEngineModeGlobal_add
cluster_test.go:1245: Step 2/2, expected an error with pattern, no match on: Error running apply: exit status 1
Error: existing RDS Clusters cannot be added to an existing RDS Global Cluster
with aws_rds_cluster.test,
on terraform_plugin_test.tf line 8, in resource "aws_rds_cluster" "test":
8: resource "aws_rds_cluster" "test" {
testing_new.go:82: Error running post-test destroy, there may be dangling resources: exit status 1
Error: removing RDS DB Cluster () from Global Cluster (tf-acc-test-5887122041588776893): InvalidParameterValue: Db cluster identifier  is not a valid arn
  status code: 400, request id: 2d998bee-adca-48f6-8317-116bc55576ef
--- FAIL: TestAccRDSCluster_GlobalClusterIdentifierEngineModeGlobal_add (157.55s)
```

### References

https://github.com/hashicorp/terraform-provider-aws/blob/e62e9c4f758bd57c9dfe8c5ed38c8eff1d107b36/internal/service/rds/global_cluster.go#L284-L343

### Would you like to implement a fix?

None

--- a/internal/service/rds/cluster_test.go
+++ b/internal/service/rds/cluster_test.go
@@ -1257,7 +1257,7 @@ func TestAccRDSCluster_GlobalClusterIdentifierEngineModeGlobal_add(t *testing.T)
 			},
 			{
 				Config:      testAccClusterConfig_GlobalClusterID_EngineMode_global(rName),
-				ExpectError: regexp.MustCompile(`Existing RDS Clusters cannot be added to an existing RDS Global Cluster`),
+				ExpectError: regexp.MustCompile(`(?i)Existing RDS Clusters cannot be added to an existing RDS Global Cluster`),
 			},
 		},
 	})
@@ -1319,7 +1319,7 @@ func TestAccRDSCluster_GlobalClusterIdentifierEngineModeGlobal_update(t *testing
 			},
 			{
 				Config:      testAccClusterConfig_GlobalClusterID_EngineMode_globalUpdate(rName, globalClusterResourceName2),
-				ExpectError: regexp.MustCompile(`Existing RDS Clusters cannot be migrated between existing RDS Global Clusters`),
+				ExpectError: regexp.MustCompile(`(?i)Existing RDS Clusters cannot be migrated between existing RDS Global Clusters`),
 			},
 		},
 	})

--- a/internal/service/rds/global_cluster.go
+++ b/internal/service/rds/global_cluster.go
@@ -285,60 +285,63 @@ func resourceGlobalClusterDelete(ctx context.Context, d *schema.ResourceData, me
 		log.Printf("[DEBUG] Removing cluster members from  RDS Global Cluster: %s", d.Id())
 
 		// The writer cluster must be removed last
+		globalClusterMembers := d.Get("global_cluster_members").(*schema.Set)
 		var writerARN string
-		for _, globalClusterMemberRaw := range d.Get("global_cluster_members").(*schema.Set).List() {
-			globalClusterMember, ok := globalClusterMemberRaw.(map[string]interface{})
+		if globalClusterMembers.Len() > 0 {
+			for _, globalClusterMemberRaw := range globalClusterMembers.List() {
+				globalClusterMember, ok := globalClusterMemberRaw.(map[string]interface{})
 
-			if !ok {
-				continue
-			}
+				if !ok {
+					continue
+				}
 
-			dbClusterArn, ok := globalClusterMember["db_cluster_arn"].(string)
+				dbClusterArn, ok := globalClusterMember["db_cluster_arn"].(string)
 
-			if !ok {
-				continue
-			}
+				if !ok {
+					continue
+				}
 
-			if globalClusterMember["is_writer"].(bool) {
-				writerARN = dbClusterArn
-				continue
+				if globalClusterMember["is_writer"].(bool) {
+					writerARN = dbClusterArn
+					continue
+				}
+
+				input := &rds.RemoveFromGlobalClusterInput{
+					DbClusterIdentifier:     aws.String(dbClusterArn),
+					GlobalClusterIdentifier: aws.String(d.Id()),
+				}
+
+				_, err := conn.RemoveFromGlobalClusterWithContext(ctx, input)
+
+				if tfawserr.ErrMessageContains(err, "InvalidParameterValue", "is not found in global cluster") {
+					continue
+				}
+
+				if err != nil {
+					return sdkdiag.AppendErrorf(diags, "removing RDS DB Cluster (%s) from Global Cluster (%s): %s", dbClusterArn, d.Id(), err)
+				}
+
+				if err := waitForGlobalClusterRemoval(ctx, conn, dbClusterArn, deadline.Remaining()); err != nil {
+					return sdkdiag.AppendErrorf(diags, "waiting for RDS DB Cluster (%s) removal from RDS Global Cluster (%s): %s", dbClusterArn, d.Id(), err)
+				}
 			}
 
 			input := &rds.RemoveFromGlobalClusterInput{
-				DbClusterIdentifier:     aws.String(dbClusterArn),
+				DbClusterIdentifier:     aws.String(writerARN),
 				GlobalClusterIdentifier: aws.String(d.Id()),
 			}
 
 			_, err := conn.RemoveFromGlobalClusterWithContext(ctx, input)
 
-			if tfawserr.ErrMessageContains(err, "InvalidParameterValue", "is not found in global cluster") {
-				continue
-			}
-
 			if err != nil {
-				return sdkdiag.AppendErrorf(diags, "removing RDS DB Cluster (%s) from Global Cluster (%s): %s", dbClusterArn, d.Id(), err)
+				if !tfawserr.ErrMessageContains(err, "InvalidParameterValue", "is not found in global cluster") {
+					return sdkdiag.AppendErrorf(diags, "removing RDS DB Cluster (%s) from Global Cluster (%s): %s", writerARN, d.Id(), err)
+				}
 			}
 
-			if err := waitForGlobalClusterRemoval(ctx, conn, dbClusterArn, deadline.Remaining()); err != nil {
-				return sdkdiag.AppendErrorf(diags, "waiting for RDS DB Cluster (%s) removal from RDS Global Cluster (%s): %s", dbClusterArn, d.Id(), err)
+			if err := waitForGlobalClusterRemoval(ctx, conn, writerARN, deadline.Remaining()); err != nil {
+				return sdkdiag.AppendErrorf(diags, "waiting for RDS DB Cluster (%s) removal from RDS Global Cluster (%s): %s", writerARN, d.Id(), err)
 			}
-		}
-
-		input := &rds.RemoveFromGlobalClusterInput{
-			DbClusterIdentifier:     aws.String(writerARN),
-			GlobalClusterIdentifier: aws.String(d.Id()),
-		}
-
-		_, err := conn.RemoveFromGlobalClusterWithContext(ctx, input)
-
-		if err != nil {
-			if !tfawserr.ErrMessageContains(err, "InvalidParameterValue", "is not found in global cluster") {
-				return sdkdiag.AppendErrorf(diags, "removing RDS DB Cluster (%s) from Global Cluster (%s): %s", writerARN, d.Id(), err)
-			}
-		}
-
-		if err := waitForGlobalClusterRemoval(ctx, conn, writerARN, deadline.Remaining()); err != nil {
-			return sdkdiag.AppendErrorf(diags, "waiting for RDS DB Cluster (%s) removal from RDS Global Cluster (%s): %s", writerARN, d.Id(), err)
 		}
 	}
 

--- a/internal/service/rds/global_cluster.go
+++ b/internal/service/rds/global_cluster.go
@@ -285,8 +285,8 @@ func resourceGlobalClusterDelete(ctx context.Context, d *schema.ResourceData, me
 		log.Printf("[DEBUG] Removing cluster members from  RDS Global Cluster: %s", d.Id())
 
 		// The writer cluster must be removed last
-		globalClusterMembers := d.Get("global_cluster_members").(*schema.Set)
 		var writerARN string
+		globalClusterMembers := d.Get("global_cluster_members").(*schema.Set)
 		if globalClusterMembers.Len() > 0 {
 			for _, globalClusterMemberRaw := range globalClusterMembers.List() {
 				globalClusterMember, ok := globalClusterMemberRaw.(map[string]interface{})

--- a/internal/service/rds/global_cluster_test.go
+++ b/internal/service/rds/global_cluster_test.go
@@ -489,6 +489,29 @@ func TestAccRDSGlobalCluster_EngineVersion_auroraPostgreSQL(t *testing.T) {
 	})
 }
 
+func TestAccRDSGlobalCluster_forceDestroy(t *testing.T) {
+	ctx := acctest.Context(t)
+	var globalCluster1 rds.GlobalCluster
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	resourceName := "aws_rds_global_cluster.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(ctx, t); testAccPreCheckGlobalCluster(ctx, t) },
+		ErrorCheck:               acctest.ErrorCheck(t, rds.EndpointsID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckGlobalClusterDestroy(ctx),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGlobalClusterConfig_forceDestroy(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGlobalClusterExists(ctx, resourceName, &globalCluster1),
+					resource.TestCheckResourceAttr(resourceName, "force_destroy", "true"),
+				),
+			},
+		},
+	})
+}
+
 func TestAccRDSGlobalCluster_sourceDBClusterIdentifier(t *testing.T) {
 	ctx := acctest.Context(t)
 	var globalCluster1 rds.GlobalCluster
@@ -749,6 +772,15 @@ resource "aws_rds_global_cluster" "test" {
   global_cluster_identifier = %[2]q
 }
 `, engine, rName)
+}
+
+func testAccGlobalClusterConfig_forceDestroy(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_rds_global_cluster" "test" {
+  global_cluster_identifier = %[1]q
+  force_destroy             = true
+}
+`, rName)
 }
 
 func testAccGlobalClusterConfig_primaryEngineVersion(rName, engine, engineVersion string) string {

--- a/internal/service/rds/global_cluster_test.go
+++ b/internal/service/rds/global_cluster_test.go
@@ -345,11 +345,11 @@ func TestAccRDSGlobalCluster_EngineVersion_updateMajor(t *testing.T) {
 				),
 			},
 			{
-				Config: testAccGlobalClusterConfig_primaryEngineVersion(rName, "aurora-mysql", "5.7.mysql_aurora.2.10.2"),
+				Config: testAccGlobalClusterConfig_primaryEngineVersion(rName, "aurora-mysql", "5.7.mysql_aurora.2.11.1"),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckGlobalClusterExists(ctx, resourceName, &globalCluster2),
 					testAccCheckGlobalClusterNotRecreated(&globalCluster1, &globalCluster2),
-					resource.TestCheckResourceAttr(resourceName, "engine_version", "5.7.mysql_aurora.2.10.2"),
+					resource.TestCheckResourceAttr(resourceName, "engine_version", "5.7.mysql_aurora.2.11.1"),
 				),
 			},
 			{
@@ -422,11 +422,11 @@ func TestAccRDSGlobalCluster_EngineVersion_updateMajorMultiRegion(t *testing.T) 
 				),
 			},
 			{
-				Config: testAccGlobalClusterConfig_engineVersionUpgradeMultiRegion(rNameGlobal, rNamePrimary, rNameSecondary, "aurora-mysql", "5.7.mysql_aurora.2.10.2"),
+				Config: testAccGlobalClusterConfig_engineVersionUpgradeMultiRegion(rNameGlobal, rNamePrimary, rNameSecondary, "aurora-mysql", "5.7.mysql_aurora.2.11.1"),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckGlobalClusterExists(ctx, resourceName, &globalCluster2),
 					testAccCheckGlobalClusterNotRecreated(&globalCluster1, &globalCluster2),
-					resource.TestCheckResourceAttr(resourceName, "engine_version", "5.7.mysql_aurora.2.10.2"),
+					resource.TestCheckResourceAttr(resourceName, "engine_version", "5.7.mysql_aurora.2.11.1"),
 				),
 			},
 		},

